src/librustc/diagnostics.rs

```diff
@@ -1923,14 +1932,14 @@ From [RFC 246]:
 "##,
 
 E0395: r##"
-The value assigned to a constant expression must be known at compile time,
+The value assigned to a constant scalar must be known at compile time,
 which is not the case when comparing raw pointers. Erroneous code example:
 
 ```
 static foo: i32 = 42;
 static bar: i32 = 43;
 
-static baz: bool = { (&foo as *const i32) == (&bar as *const i32) };
+static baz: bool = { (&foo as *const i32) < (&bar as *const i32) };
```

Comment:
Probably because inequality could be checked for pointers to statics/constants whose values differ, since they obviously cannot be located at the same memory address. Equality and order cannot be checked, since the memory locations are undefined, and even constants that have the same value might be placed at different memory locations